.github/workflows/onPRClose.yml

```diff
@@ -3,41 +3,75 @@ on:
   pull_request:
     types: [closed]
 jobs:
-  extractIssueNumber:
-    name: Extract Issue Number
+
+  get-pr-number:
+    name: Get Current PR Number
+    runs-on: ubuntu-latest
+    outputs: 
+      pr-number: ${{ steps.condval.outputs.value }}
+    steps:
+      - uses: 3drepo/action-cond@v1
+        id: condval
+        with:
+          cond: ${{ github.event_name == 'pull_request' }}
+          if_true: ${{ github.event.number }}
+          if_false: ${{ github.event.pull_request.number }}
+
+  get-issue-number:
+    name: Get Current Issue Number
+    needs: [ get-pr-number ]
     runs-on: ubuntu-latest
     outputs: 
       issue-number: ${{ steps.commitMsgParser.outputs.issue-number }}
+      content-id: ${{ steps.commitMsgParser.outputs.content-id }}
     steps:
       - name: Extract Issue Number from PR
-        uses: carmenfan/extract-issue-number-from-PR-merge@v1.1
+        uses: 3drepo/extract-issue-number-from-PR@v1
         with:
-          pr: ${{ github.event.number }}
+          pr: ${{ needs.get-pr-number.outputs.pr-number }}
           base: staging
           github-token: ${{ secrets.GITHUB_TOKEN }}
         id: commitMsgParser
-  tagOnPRMerge:
+
+  tag-on-pr-merge:
     name: Tag merged issues as In staging if applicable
     runs-on: ubuntu-latest
-    needs: extractIssueNumber
+    needs: get-issue-number
     steps:
       - name: Apply in staging label
-        if: needs.extractIssueNumber.outputs.issue-number
+        if: needs.get-issue-number.outputs.issue-number
```

Comment:
If I'm reading [this](https://github.com/3drepo/extract-issue-number-from-PR/blob/v1/index.js) correctly, you're extracting the `issue-number`[ regardless of whether the issue is being merged into staging](https://github.com/3drepo/extract-issue-number-from-PR/blob/v1/index.js#L72). So provided the branch name matches the regex it'll set the `issue-number` output. Meaning we will be tagging pretty much every issue as `In Staging`

So I think you need to check for the output `merged` here? [which is only provided if it's merged](https://github.com/3drepo/extract-issue-number-from-PR/blob/v1/index.js#L78) and the base matches the target base.

I didn't test it, so I could be missing something

Reply:
yes, this is because I didn't actually touch this as it was already working previously, but then I did a search and replace which changed the action to the 3D Repo one with the new behaviour to always return the issue number and merged as optional, as I needed it all the time and the previous action only returned it if it was merged. Anyway, yes, it should check merged and only fire on that.